plugin/ochttp/propagation/tracecontext/propagation.go

```diff
@@ -87,15 +90,71 @@ func (f *HTTPFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanCon
 		return trace.SpanContext{}, false
 	}
 
+	// Extract Tracestate
+	tracestate, ok := tracestateFromRequest(req)
+	if ok == false {
+		return trace.SpanContext{}, false
+	}
+	sc.Tracestate = tracestate
 	return sc, true
 }
 
-// SpanContextToRequest modifies the given request to include a header.
+func tracestateFromRequest(req *http.Request) (*tracestate.Tracestate, bool) {
+	h := req.Header.Get(tracestateHeader)
+	if h == "" {
+		return nil, true
+	}
+
+	var entries []tracestate.Entry
+	pairs := strings.Split(h, ",")
+	headerLenWithoutTrailingSpaces := len(pairs) - 1 // Number of commas
+	for _, pair := range pairs {
+		trimmedPair := strings.TrimSpace(pair)
+		headerLenWithoutTrailingSpaces += len(trimmedPair)
+		if headerLenWithoutTrailingSpaces > maxTracestateLen {
+			// Drop the entire Tracestate
+			return nil, true
+		}
+		kv := strings.Split(strings.TrimSpace(pair), "=")
```

Comment:
What about using strings.Index here? It should be cheaper than a split.

Reply:
I need the split  for creating Tracesate.Entry couple of lines later. 
entries = append(entries, tracestate.Entry{Key: kv[0], Value: kv[1]})